scipy/special/meson.build

```diff
@@ -283,6 +208,14 @@ py3.extension_module('_specfun',
   subdir: 'scipy/special'
 )
 
+py3.extension_module('_cdflib',
+  cython_gen.process('_cdflib.pyx'),
```

Comment:
This should be moved to above the `extension_module` call and changed to:
```meson
_cdflib_c = cython_gen.process('_cdflib.pyx')
```
And then `_cdflib_c` (no quotes) should be used as the source object for `_ufuncs` and `cython_special`. That avoids re-running Cython on the same file 3 times:

```diff
diff --git a/scipy/special/meson.build b/scipy/special/meson.build
index 334fba938..fbfea0c0a 100644
--- a/scipy/special/meson.build
+++ b/scipy/special/meson.build
@@ -208,8 +208,9 @@ py3.extension_module('_specfun',
   subdir: 'scipy/special'
 )
 
+_cdflib_c = cython_gen.process('_cdflib.pyx')
 py3.extension_module('_cdflib',
-  cython_gen.process('_cdflib.pyx'),
+  _cdflib_c,
   c_args: numpy_nodepr_api,
   dependencies: [np_dep, npymath_lib],
   install: true,
@@ -261,7 +262,7 @@ py3.extension_module('_ufuncs',
   [
     ufuncs_sources,
     uf_cython_gen.process(cython_special[0]),  # _ufuncs.pyx
-    '_cdflib.pyx'
+    _cdflib_c,
   ],
   c_args: [cython_c_args, Wno_maybe_uninitialized],
   include_directories: ['../_lib', '../_build_utils/src'],
@@ -333,7 +334,7 @@ py3.extension_module('_ellip_harm_2',
 py3.extension_module('cython_special',
   [
     uf_cython_gen.process(cython_special[6]),  # cython_special.pyx
-    '_cdflib.pyx',
+    _cdflib_c,
     '_cosine.c',
     'scaled_exp1.c',
     'amos_wrappers.c',
```

Reply:
Ah great! That's indeed very helpful.